.github/workflows/rules.yml

```diff
@@ -0,0 +1,73 @@
+name: Rules 
+
+on:
+
+  workflow_call:
+    inputs:
+        os:
+          required: true 
+          type: string 
+        python-version:
+          required: true 
+          type: string
+        environment:
+          required: true 
+          type: string 
+    secrets:
+      CODECOV_TOKEN:
+        required: true
+
+jobs:
+  build:
+    runs-on: ${{ inputs.os }}
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ inputs.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ inputs.python-version }}
+    - name: Install package
+      run: |
+        python -m pip install --upgrade pip
+        pip install pytest-cov
+        pip install pylint
+        echo "${{ inputs.environment }}"
+        if [[ "${{ inputs.environment }}" == "qibo" ]]; then
+          pip install .[tests,docs]
+          pip install git+https://github.com/qiboteam/qibojit
+          pip install tensorflow
+        fi
+    - name: Test with pylint
+      run: |
+        if [[ "${{ inputs.environment  }}" == "qibo" ]]; then
+          pylint src/**/*.py -E -f colorized
+          pylint src/**/*.py --exit-zero -f colorized
+        fi
+        if [[ "${{ inputs.environment  }}" == "qibolab" ]]; then
+          pylint src -E -d E1123,E1120
+        fi
+      - name: Test with pytest core
+        run: |
+          if [[ "${{ inputs.environment  }}" == "qibo" ]]; then
+            pytest src/qibo/tests/ --cov=qibo --cov-report=xml --pyargs qibo --durations=60
+          fi
+          if [[ "${{ inputs.environment  }}" == "qibolab" ]]; then
+            pytest src/qibolab --cov=qibolab --cov-report=xml --pyargs qibolab -m "not qpu" --platforms tii1q_b1,qw5q_gold
+          fi
+      - name: Test example 
+        if: ${{ inputs.environment == 'qibo'  }} && ${{ inputs.os == 'ubuntu' }}
```

Comment:
Since the content is not Qibo-specific, you can just add a `inputs.doctest` boolean, to activate/deactivate this step (you make it `false` by default, and move the condition on Ubuntu in the Qibo caller)